src/Util/ReactionUtil.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * EdwardML Reactions
+ *
+ * Using the reactions system, the user can 'react' to content.  The content
+ * receiving a reaction is called the 'target'.  There are different types of
+ * reactions, and the user can select to perform any subset of them on a
+ * specific target.
+ *
+ * Key concepts:
+ *
+ * - UserRxns: a document holding the user's reaction states toward a target.
+ *
+ * - CountDoc: a document holding current reaction counts for a target.
+ *
+ * - RxnTarget: An object holding a description of the target of a reaction.
+ *       Should be obtained ONLY via functions in this file.
+ *
+ * - RxnType: An object describing a type of reaction.  Should be obtained
+ *       ONLY from const `RxnTypes` exported from this file.
+ */
+
+import { HandsClapping, Heart, Trash } from "phosphor-react";
+
+/**
+ * @typedef {Object} RxnTarget
+ * @property {"review"|"list"|"listComment"} type The type of target.
+ * @property {string} noun A noun to use in UI text.
+ * @property {string} entityId A globally unique ID for this reaction target.
+ * @property {string} countDocPath A path to the document with reaction counts.
+ * @property {string} ownerId The owner of this target.  They may be notified.
+ */
+
+/**
+ * Build a `RxnTarget` for an anime review.
+ * @param {number} animeId The id of the anime being reviewed.
+ * @param {string} reviewAuthorId The uid of the author of the reivew.
+ * @returns {RxnTarget} A `RxnTarget` for the specified anime review.
+ */
+export function getRxnTargetForAnimeReview(animeId, reviewAuthorId) {
+  return {
+    type: "review",
+    noun: "review",
+    entityId: `review-${animeId}-${reviewAuthorId}`,
+    countDocPath: ["animeData", animeId, "reviews", reviewAuthorId].join("/"),
+    ownerId: reviewAuthorId,
+  };
+}
+
+/**
+ * Build a `RxnTarget` for a watchlist.
+ * @param {string} listId The id of the list.
+ * @param {string} listOwnerId The uid of the owner of the list.
+ * @returns {RxnTarget} A `RxnTarget` for the specified watchlist.
+ */
+export function getRxnTargetForWatchlist(listId, listOwnerId) {
+  return {
+    type: "list",
+    noun: "list",
+    entityId: `list-${listOwnerId}${listId}`,
+    countDocPath: ["watchlistData", listOwnerId + listId].join("/"),
+    ownerId: listOwnerId,
+  };
+}
+
+/**
+ * Build a `RxnTarget` for a watchlist comment.
+ * @param {string} listId The id of the list being commented on.
+ * @param {string} listOwnerId The uid of the owner of the list.
+ * @param {string} commentAuthorId The uid of the author of the comment.
+ * @returns {RxnTarget} A `RxnTarget` for the specified watchlist comment.
+ */
+export function getRxnTargetForWatchlistComment(
+  listId,
+  listOwnerId,
+  commentAuthorId
+) {
+  return {
+    type: "listComment",
+    noun: "comment",
+    entityId: `listComment-${listOwnerId}${listId}-${commentAuthorId}`,
+    countDocPath: [
+      "watchlistData",
+      listOwnerId + listId,
```

Comment:
The `listId` variable already includes the `listOwnerId`, so we'll need to delete `listOwnerId` here.

Reply:
Oh, nice catch!  I think we need a name for the id that is formed when we smush together the list owner id with the list id -- maybe `userListId` or something like that

Reply:
I've actually chosen to keep this function signature requiring both `listId` and `listOwnerId` separately, since I ended up needing them separate in the notification.  I've just updated the call to this method in `Review` to pass in a `listId`, which I piped through from `ReviewContainer`.  Lmk what you think